<h1>Ken Lloyd O. Alcoreza</h1>

<h1>Data Analyst</h1>
<hr>
<h3>Data: Ken Lloyd Alcoreza</h3>

In [1]:
import numpy as np
import pandas as pd



In [2]:
panda = pd.read_csv('datasets\\penguins.csv')

In [3]:
panda

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [4]:
panda.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
panda.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [6]:
panda.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [7]:
panda['species']. unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [8]:
panda['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [9]:
panda['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [10]:
panda['sex'].value_counts()

sex
male      168
female    165
Name: count, dtype: int64

In [11]:
chin = panda[panda['species']=='chinstrap']

In [12]:
chins = panda[panda ['body_mass_g']>=4000]

In [13]:
chan = (panda ['body_mass_g']>=4000) & (panda['species']=='chinstrap')

In [14]:
chan.info()

<class 'pandas.core.series.Series'>
RangeIndex: 344 entries, 0 to 343
Series name: None
Non-Null Count  Dtype
--------------  -----
344 non-null    bool 
dtypes: bool(1)
memory usage: 476.0 bytes


In [15]:
chano = (panda ['body_mass_g']>=4000) & (panda['species']=='chinstrap') & (panda['sex']=='female')

In [16]:
chano.info()

<class 'pandas.core.series.Series'>
RangeIndex: 344 entries, 0 to 343
Series name: None
Non-Null Count  Dtype
--------------  -----
344 non-null    bool 
dtypes: bool(1)
memory usage: 476.0 bytes


In [17]:
panda.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [18]:
panda['bill_length_mm'] = panda['bill_length_mm'].fillna(panda['bill_length_mm'].mean)
panda['bill_depth_mm'] = panda['bill_depth_mm'].fillna(panda['bill_depth_mm'].mean)
panda['flipper_length_mm'] = panda['flipper_length_mm'].fillna(panda['flipper_length_mm'].mean)
panda['body_mass_g'] = panda['body_mass_g'].fillna(panda['body_mass_g'].mean)
panda['sex'] = panda['sex'].fillna(panda['sex'].mode()[0])




In [19]:
panda.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h1>Exploratory Data Analytics (EDA</h1>

<h2>1.Which has Species of penguin has the most data?</h2>

In [20]:
panda['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h1>Insight</h1>
<h2>The Species of penguin that has the most data is Adelie </h2>

<h1>2. What are the species of the penguins that are recorded?</h1>


In [21]:
panda['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

<h1>Insight:</h1>
<h2>There are Adeleie, Gentoo, and Chinstrap Species of a penguin</h2>

<h1>3. In males, what is the most average flipper lenght?</h1>

In [22]:
ml_fl = panda.groupby([panda['sex'] == 'Male']).count()
ml_fl['flipper_length_mm'].mean()


np.float64(344.0)

<h1>Insight:</h1>
<h2>The Average Flipper Length of the males is 344</h2>

<h1>4. The average flipper length of females?</h1>

In [23]:
ml_fl = panda[panda['sex'] == 'female'].count()
ml_fl['flipper_length_mm'].mean()

np.float64(165.0)

<h1>Insight:</h1>
<h2>Tghe Average flipper Length of Females is 165</h2>

<h1>5. What are the island found on the data</h1>

In [24]:
panda['island'].unique()

array(['Torgersen', 'Biscoe', 'Dream'], dtype=object)

<h1>Insight:</h1>
<h2>Island in the data are Torgesen,Biscoe and Dream</h2>

<h1>6. The average Beak Length of the Adelie?</h1>

In [25]:
avg_adl_bill = panda.groupby([panda['species'] == 'adelie']).count()
avg_adl_bill['flipper_length_mm'].mean()

np.float64(344.0)

<h1>Insight:</h1>
<h2>The Average beak length of adelie is 344</h2>

<h1>7.In the island of torgersen, how many males are in there?</h1>

In [26]:
ml_tor = panda[(panda['island'] == 'Torgersen')]
ml_tor['sex'].value_counts()

sex
male      28
female    24
Name: count, dtype: int64

<h1>Insight:</h1>
<h2>the count of males on the island Torgersen is 28</h2>

<h1>8. Data Recorded of the females on the Island Biscoe</h1>

In [27]:
ml_tor = panda[(panda['island'] == 'Biscoe')]
ml_tor['sex'].value_counts()

sex
male      88
female    80
Name: count, dtype: int64

<h1>Insight:</h1>
<h2>Data of females on the island Biscoe is 80</h2>

<h1>9. How many penguins are on Biscoe and Torgersen combined?</h1>

In [28]:
p_bis = panda['island'] == 'Biscoe'
p_tor = panda['island'] == 'Torgersen'

p_tor.sum() + p_bis.sum()

np.int64(220)

<h1>Insight:</h1>
<h2>The combined data from the island Biscoe and Torgersen has the sum of 220</h2>

<h1>10. What is year has the highest data recorded?</h1>

In [29]:
panda['year'].unique()

array([2007, 2008, 2009])

In [30]:
 (panda['year'] == 2007).sum()


np.int64(110)

In [31]:
 (panda['year'] == 2008).sum()


np.int64(114)

In [32]:
 (panda['year'] == 2009).sum()


np.int64(120)

<h1>Insight:</h1>
<h2>The year that has the most data is the year 2009</h2>

<h1>11. What is the average male flipper length on the island Biscoe?</h1>

In [33]:
avg_fl_m = panda[(panda['sex'] == 'male') & (panda['island'] == 'Biscoe')]
avg_fl_m['flipper_length_mm'].mean()


TypeError: unsupported operand type(s) for +: 'float' and 'method'

<h1>12. What is the Average of the beak length of all the penguin?</h1>

In [34]:
panda['bill_length_mm'].mean()

TypeError: unsupported operand type(s) for +: 'float' and 'method'

<h1>13. What is the average flipper length of all the penguin?</h1>

In [36]:
panda[panda['flipper_length_mm']].mean()

KeyError: "None of [Index([                                                                                                                                                                                                                                       181.0,\n                                                                                                                                                                                                                                              186.0,\n                                                                                                                                                                                                                                              195.0,\n       <bound method Series.mean of 0      181.0\\n1      186.0\\n2      195.0\\n3        NaN\\n4      193.0\\n       ...  \\n339    207.0\\n340    202.0\\n341    193.0\\n342    210.0\\n343    198.0\\nName: flipper_length_mm, Length: 344, dtype: float64>,\n                                                                                                                                                                                                                                              193.0,\n                                                                                                                                                                                                                                              190.0,\n                                                                                                                                                                                                                                              181.0,\n                                                                                                                                                                                                                                              195.0,\n                                                                                                                                                                                                                                              193.0,\n                                                                                                                                                                                                                                              190.0,\n       ...\n                                                                                                                                                                                                                                              202.0,\n                                                                                                                                                                                                                                              194.0,\n                                                                                                                                                                                                                                              206.0,\n                                                                                                                                                                                                                                              189.0,\n                                                                                                                                                                                                                                              195.0,\n                                                                                                                                                                                                                                              207.0,\n                                                                                                                                                                                                                                              202.0,\n                                                                                                                                                                                                                                              193.0,\n                                                                                                                                                                                                                                              210.0,\n                                                                                                                                                                                                                                              198.0],\n      dtype='object', length=344)] are in the [columns]"